trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/ErrorResponse.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.rest;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+import java.util.Optional;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JacksonXmlRootElement(localName = "Error")
+public record ErrorResponse(String code, String message, String resource, Optional<String> requestId)
+{ }
```

Comment:
Please add compact constructor with notNull checks

Reply:
Some of them may be null though right? They're being excluded with the `@JsonInclude`annotation

Reply:
Happy to maintain null safety, but we should then make sure we never construct one of these with a null value. We don't want to have our exception handling logic result in an unhandled exception because of a null check when constructing the response